iotc-generic-c-sdk/include/iotconnect.h

```diff
@@ -64,10 +64,12 @@ typedef struct {
     IotclCommandCallback cmd_cb; // callback for command events.
     IotclMessageCallback msg_cb; // callback for ALL messages, including the specific ones like cmd or ota callback.
     IotConnectStatusCallback status_cb; // callback for connection status
+    bool dynamic_memory;
 } IotConnectClientConfig;
 
 
 IotConnectClientConfig *iotconnect_sdk_init_and_get_config(void);
+IotConnectClientConfig *iotconnect_sdk_init_and_alloc_config(char *env, char *cpid, char *duid, int authentication_type, char *symmetric_key);
```

Comment:
This piece is probably my biggest gripe with the change. We are are pretty much saying that we want to let the SDK clone each and every value that we configured, but how hard is it actually for the client to do this and why aren't we then cloning the cert and key in case of x509? The answer is we aren't cloning it becasue it's usually wasteful. The cert will probably be somewhere in permanent RAM storage. This brings up the previous point that the client knows best which storage needs cloning and which does not.

Anothe rproblem is that these areguments are positional. We should just accept the client's struct instead so that we don't have to change the signature every time we change the config.

Now PC's have plenty of RAM and we could really get away with cloning the config fully. So the client gets an all or nothing, which could be helpful in some cases. All in all, perhaps we could modify the signature and name and also be specific in comments:
```c
// The user is responsible for maintaining string references around while SDK is running. The client should set up the returned struct before calling iotconnect_sdk_init()
IotConnectClientConfig *iotconnect_sdk_init_and_get_config(void);

// The SDK will clone each string provided in the config and maintain references while the SDK is running. The client should pass a struct that the SDK will deep copy. The use is responsible for chacking the return value which will be NULL if alloccation has failed.
IotConnectClientConfig *iotconnect_sdk_init_deep_copy(IotConnectClientConfig * c);
```